flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/MultiInputTransformationTranslator.java

```diff
@@ -54,7 +55,11 @@
 		Collection<Integer> ids = translateInternal(transformation, context);
 		boolean isKeyed = transformation instanceof KeyedMultipleInputTransformation;
```

Comment:
I think this check is not sufficient now. We should check if any of the inputs has a state selector.